File: 0283-move-zeroes/0283-move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int non_zero=0;
        for(int i=0;i<nums.size();i++)
        {
            if(nums[i]!=0)
            {
                swap(nums[non_zero],nums[i]);
                non_zero++;
            }
        }
        
    }
};

File: 0283-move-zeroes/0283-move-zeroes.py
class Solution:
    def moveZeroes(self, nums: list[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        non_zero=0
        n=len(nums)
        for i in range(n):
            if nums[i]!=0:
                nums[non_zero],nums[i]=nums[i],nums[non_zero]
                non_zero+=1
        